### PR TITLE
Add new Dart 2 Map, Iterable, and List methods to multimap stuff

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: lib/src/collection/multimap.dart [hunk 1]
```diff
@@ -309,51 +309,44 @@ class _WrappedMap<K, V, C extends Iterable<V>> implements Map<K, C> {
...
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_getter
-  Iterable<Null> get entries {
-    // Change Iterable<Null> to Iterable<MapEntry<K, V>> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("entries");
+  Iterable<MapEntry<K, C>> get entries {
+    return keys.map((K key) => new MapEntry<K, C>(key, this[key]));
```
Comment (a14n) [inline lib/src/collection/multimap.dart]: Perhaps:
```dart
Iterable<MapEntry<K, C>> get entries => _multimap._map.entries;
```

File: lib/src/collection/multimap.dart [hunk 2]
```diff
@@ -309,51 +309,44 @@ class _WrappedMap<K, V, C extends Iterable<V>> implements Map<K, C> {
...
-    // Change Object to MapEntry<K2, V2> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("map");
+  Map<K2, C2> map<K2, C2>(MapEntry<K2, C2> transform(K key, C value)) {
+    var result = <K2, C2>{};
+    for (var key in this.keys) {
+      var entry = transform(key, this[key]);
+      result[entry.key] = entry.value;
+    }
+    return result;
```
Comment (a14n) [inline lib/src/collection/multimap.dart]:
```dart
  Map<K2, C2> map<K2, C2>(MapEntry<K2, C2> transform(K key, C value)) =>
      _multimap._map.map(transform);
```

File: lib/src/collection/multimap.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/collection/multimap.dart]: Done.

File: lib/src/collection/multimap.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/collection/multimap.dart]: Done.

File: lib/src/collection/multimap.dart [hunk 3]
```diff
@@ -309,51 +309,36 @@ class _WrappedMap<K, V, C extends Iterable<V>> implements Map<K, C> {
...
-  }
+  Map<K2, C2> map<K2, C2>(MapEntry<K2, C2> transform(K key, C value)) =>
+      _multimap._map.map(transform);
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
   C update(K key, C update(C value), {C ifAbsent()}) {
-    throw new UnimplementedError("update");
+    throw new UnsupportedError("Update unsupported on map view");
```
Comment (cbracken) [inline lib/src/collection/multimap.dart]: Can you add a TODO here and below? This seems like it should be easy to support.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,16 @@
      `followedBy`, and accepts the `orElse` parameter on `singleWhere`.
    * New: DelegatingList now includes real implementations of `operator +`,
      `indexWhere`, and `lastIndexWhere`.
+   * New: The map returned by `Multimap.asMap()` now includes real
+     implementations of `get entries` and `removeWhere`. This class also has
+     "real" implementations of `addEntries`, `map`, `update`, and `updateAll`,
+     which just throw an `UnsupportedError`, as inserts and updates are not
+     allowed on map views.
```
Comment (cbracken) [inline CHANGELOG.md]: Why are they not allowed? We currently allow updates on the existing map view.

**Comment by cbracken** [comment]
Partial fix to #418.

**Pull request merged**